sql/updates/world/master/2022_99_99_99_quest_34586.sql

```diff
@@ -0,0 +1,14 @@
+UPDATE `creature_template` SET `scriptname` = 'npc_baros_alexston' WHERE `entry` = 79243;
+
+-- TerrainSwap
+DELETE FROM `terrain_swap_defaults` WHERE (`MapId` = 1116 AND `TerrainSwapMap` = 1158);
+INSERT INTO `terrain_swap_defaults` (`MapId`, `TerrainSwapMap`, `Comment`) VALUES
+(1116, 1158, 'SMV Alliance Garrison Level 1');
+
+-- Condition
+DELETE FROM `conditions` WHERE (`SourceTypeOrReferenceId` = 25 AND `SourceGroup` = 0 AND `SourceEntry` = 1158) OR (`SourceTypeOrReferenceId` = 13 AND `SourceGroup` = 1 AND `SourceEntry` = 160938);
+INSERT INTO `conditions` (`SourceTypeOrReferenceId`, `SourceGroup`, `SourceEntry`, `SourceId`, `ElseGroup`, `ConditionTypeOrReference`, `ConditionTarget`, `ConditionValue1`, `ConditionValue2`, `ConditionValue3`, `NegativeCondition`, `Comment`) VALUES
+(25, 0, 1158, 0, 0, 47, 0, 34586, 66, 0, 0, 'TerrainSwap to 1158 if Quest: 34586 is completed/rewarded'),
```

Comment:
Just a hint about this condition: apparently SPELL_CREATE_GARRISON_SHADOWMOON_VALLEY_ALLIANCE's effect 1 is 
![image](https://user-images.githubusercontent.com/30064812/175830998-4d580303-4daf-4997-ad35-d7d44b01377e.png)

If you check Wowhead, you can see there's a hidden quest being tracked Founded Garrison (https://www.wowhead.com/quest=36797/founded-garrison). I'd take a guess and say this is the correct condition being used for the terrain swap.

Reply:
Oh yes that could be. Hidden Quest is also used in the Blasted Lands TerrainSwap. Thanks for the Hint. :D